olp-cpp-sdk-dataservice-read/include/olp/dataservice/read/model/Messages.h

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <olp/dataservice/read/DataServiceReadApi.h>
+#include <olp/dataservice/read/model/Data.h>
+#include <olp/dataservice/read/model/StreamOffsets.h>
+#include <boost/optional.hpp>
+
+namespace olp {
+namespace dataservice {
+namespace read {
+namespace model {
+
+/**
+ * @brief Encapsulates information about actual data content.
+ */
+class DATASERVICE_READ_API Metadata final {
+ public:
+  /**
+   * @brief Gets the partition of this metata.
+   *
+   * @return The
+   * [partition](https://developer.here.com/olp/documentation/data-api/data_dev_guide/rest/layers/partitions.html)
```

Comment:
This is not entirely correct as we only talk about Stream Layer Partition which might be totally different from the TileKey partitions commonly used by OLP. Also this Metadata is not the common metadata returned by Query/Meta API.

Reply:
Agreed to enhance this comment later.